llama_index/llms/openai_utils.py

```diff
@@ -107,29 +110,45 @@
 CompletionClientType = Union[Type[Completion], Type[ChatCompletion]]
 
 
-def _create_retry_decorator(max_retries: int) -> Callable[[Any], Any]:
+def create_retry_decorator(
+    max_retries: int,
+    exponential: bool = True,
+    stop_after_delay_seconds: Union[int, None] = None,
```

Comment:
Let's use `Optional` over `Union[int, None]`

Reply:
Also, can the time be a `float`?  I think `float` is possible

Reply:
<img width="727" alt="Screenshot 2023-11-01 at 1 11 50 PM" src="https://github.com/run-llama/llama_index/assets/108733252/66ff8112-c2b5-4c53-8aa2-8809986e536c">
<img width="465" alt="Screenshot 2023-11-01 at 1 12 28 PM" src="https://github.com/run-llama/llama_index/assets/108733252/79178800-4344-48e3-94b4-bda8bd13ccf3">
Yep @jamesbraza looks like float is possible.